forms-flow-api/src/formsflow_api/models/application.py

```diff
@@ -881,11 +879,11 @@ def apply_submission_sorting(cls, query, sort_by, sort_order, submission_alias):
         """Sort by submission count or latest submission time."""
         order_func = desc if "desc" in sort_order else asc
         if "latestSubmission" in sort_by:
-            # Sort by latest submission time, treating NULLs as max datetime
+            # Sort by latest submission time, treating NULLs first while ascending & last while descending
             return query.order_by(
-                order_func(
-                    func.coalesce(submission_alias.c.latest_submission, datetime.max)
-                )
+                order_func(submission_alias.c.latest_submission).nulls_last()
+                if order_func == desc
+                else order_func(submission_alias.c.latest_submission).nulls_first()
```

Comment:
is it possible to do something like
`sort_criteria = nulls_last(desc(submission_alias.c.latest_submission)) if "desc" in sort_order else nulls_first(asc(...))` and do `query.order_by(sort_criteria)` ? We can avoid checking same condition twice.
Or set both `order_func` and `null_func` in variables and use `query.order_by(null_func(order_fun(...)))`.

Reply:
Done